src/sql/workbench/contrib/charts/browser/graphInsight.ts

```diff
@@ -159,28 +165,33 @@ export class Graph implements IInsight {
 			return mixin(c, getColors(this.options.type, i, c.data!.length), false);
 		});
 
+		if (this.options.type === 'horizontalBar') {
+			this.options.type = ChartType.Bar;
+			this.options.indexAxis = 'y';
+		}
 		if (this.chartjs) {
 			this.chartjs.data.datasets = chartData;
-			this.chartjs.config.type = this.options.type;
+			(<chartjs.ChartConfiguration>this.chartjs.config).type = ChartTypeToChartJsType[this.options.type]
```

Comment:
It is a union of 2 different types where the second type does not contain 'type' property. Is there a better way to do it without casting?

Reply:
Gotcha, little odd but not a huge deal then. 

There isn't really a clean way to do this - we can check the type using a [type guard function](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) but that's a bit overkill since we can just know that we always specify a type ourselves (and since it accepts both types their logic should be able to handle it regardless)